components/gitpod-db/src/typeorm/project-db-impl.ts

```diff
@@ -33,15 +33,35 @@ export class ProjectDBImpl implements ProjectDB {
         return repo.findOne({ cloneUrl, markedDeleted: false });
     }
 
-    public async findProjectsByCloneUrls(cloneUrls: string[]): Promise<Project[]> {
+    public async findProjectsByCloneUrls(cloneUrls: string[]): Promise<(Project & { owners?: string[] })[]> {
```

Comment:
This isn't actually "project owners", but "team owners" (e.g. maybe some team owners have nothing to do with this actual project, and also we never add `p.userId`).

Please remove the confusion by renaming `owners` to `teamOwners`:

```suggestion
    public async findProjectsByCloneUrls(cloneUrls: string[]): Promise<(Project & { teamOwners?: string[] })[]> {
```